samples/CommunityToolkit.Maui.Sample/AppShell.xaml.cs

```diff
@@ -12,62 +14,63 @@ namespace CommunityToolkit.Maui.Sample;
 
 public partial class AppShell : Shell
 {
-	static readonly IReadOnlyDictionary<Type, (Type GalleryPageType, Type ContentPageType)> viewModelMappings = new Dictionary<Type, (Type, Type)>()
+	static readonly IReadOnlyDictionary<Type, (Type GalleryPageType, Type ContentPageType)> viewModelMappings = new Dictionary<Type, (Type, Type)>(new[]
 	{
 		// Add Alerts View Models
-		{ typeof(SnackbarViewModel), (typeof(AlertsGalleryPage), typeof(SnackbarPage)) },
-		{ typeof(ToastViewModel), (typeof(AlertsGalleryPage), typeof(ToastPage)) },
+		CreateViewModelMapping<SnackbarPage, SnackbarViewModel, AlertsGalleryPage, AlertsGalleryViewModel>(),
```

Comment:
AlertGalleryViewModel is not necessary here. This mapping is used to build the path for Shell

Reply:
You're totally right! 

But I couldn't figure out a way to constrain `TGalleryPage` for `CreateViewModelMapping` without also requiring + constraining `TGalleryViewModel `.

The reason I created the `CreateViewModelMapping` method is to ensure that every new PR implements the following correctly:
- [x] Sample Page implements `BasePage<T>`
- [x] Sample View Models implement `BaseViewModel` 
- [x] Sample Gallery Pages implement `BaseGalleryPage<T>` 

To ensure this, I implemented the following constraints on `CreateViewModelMapping`:
- `where TPage : BasePage<TViewModel>`
- `where TViewModel : BaseViewModel`
- `where TGalleryPage : BaseGalleryPage<TGalleryViewModel>`

But in order to constrain `TGalleryPage` to `BaseGalleryPage<T>`, the constraints also needs to know that `T` is constrained to `BaseGalleryViewModel`. 

And when I try to use `where TGalleryPage : BaseGalleryPage<BaseGalleryViewModel>`, I get this error:

| `CreateViewModelMapping` without `TGalleryViewModel` | Error | 
| ---------------------------------------------------------- | ----- |
| <img width="863" alt="Screen Shot 2022-03-05 at 12 08 35 PM" src="https://user-images.githubusercontent.com/13558917/156898529-209353f2-f08c-4f59-9054-eb324f07f4a8.png"> | <img width="1063" alt="image" src="https://user-images.githubusercontent.com/13558917/156898619-1f97cde9-059e-42f8-b457-a98f21bee5f4.png"> |


So, long story short, I couldn't figure out a way to constrain `TGalleryPage` without also including `TGalleryViewModel`.

I'm totally down to remove `TGalleryViewModel` if you can figure out how to implement `CreateViewModelMapping` without it, though!